File: code401challenges/src/main/java/code401challenges/utilities/MultiBracketValidation.java
package code401challenges.utilities;

import code401challenges.stacksAndQueues.Stack;

public class MultiBracketValidation {

    public static boolean multiBracketValidation(String input) {
        Stack round = new Stack();;
        Stack curly = new Stack();;
        Stack square = new Stack();;

        for (int i = 0; i < input.length(); i++) {
            char c = input.charAt(i);

            if (c == '(') {
                if (round.isEmpty()) {
                    round.push("(");
                } else { return false; }
            } else if ( c == '{') {
                if (curly.isEmpty()) {
                    curly.push("{");
                } else { return false; }
            } else if (c == '[') {
                if (square.isEmpty()) {
                    square.push("[");
                } else { return false; }
            } else if (c == ')') {
                if (!round.isEmpty()) {
                    round.pop();
                } else { return false; }
            } else if (c == '}') {
                if (!curly.isEmpty()) {
                    curly.pop();
                } else { return false; }
            } else if (c == ']') {
                if (!square.isEmpty()) {
                    square.pop();
                } else { return false; }
            }
        }

        if (round.isEmpty() && curly.isEmpty() && square.isEmpty()) {
            return true;
        } else {
            return false;
        }
    }
}

File: js401challenges/Stacks_Queues/MultiBracketValidation.js
'use strict';

const Stack = require('./Stack');

class MultiBracketValidation {

  constructor () {
    this.round = new Stack();
    this.curly = new Stack();
    this.square = new Stack();
  }

  validate (input) {
    for (let i = 0; i < input.length; i++) {
      let c = input.charAt(i);

      if (c === '(') {
        if (this.round.isEmpty()) {
          this.round.push('(');
        } else { return false; }
      } else if ( c === '{') {
        if (this.curly.isEmpty()) {
          this.curly.push('{');
        } else { return false; }
      } else if (c === '[') {
        if (this.square.isEmpty()) {
          this.square.push('[');
        } else { return false; }
      } else if (c === ')') {
        if (!this.round.isEmpty()) {
          this.round.pop();
        } else { return false; }
      } else if (c === '}') {
        if (!this.curly.isEmpty()) {
          this.curly.pop();
        } else { return false; }
      } else if (c === ']') {
        if (!this.square.isEmpty()) {
          this.square.pop();
        } else { return false; }
      }
    }

    if (this.round.isEmpty() && this.curly.isEmpty() && this.square.isEmpty()) {
      return true;
    } else {
      return false;
    }
  }

}

module.exports = MultiBracketValidation;